ledger/participant-integration-api/src/main/scala/platform/store/backend/postgresql/PostgresEventStrategy.scala

```diff
@@ -7,45 +7,72 @@ import com.daml.platform.store.appendonlydao.events.Party
       parties: Set[Party],
-  ): CompositeSql =
-    if (parties.size == 1)
-      cSQL"array[${parties.head.toString}]::text[]"
-    else {
-      val partiesArray: Array[String] = parties.view.map(_.toString).toArray
-      cSQL"array(select unnest(#$witnessesColumnName) intersect select unnest($partiesArray::text[]))"
-    }
+      stringInterning: StringInterning,
+  ): CompositeSql = {
+    val internedParties: Array[java.lang.Integer] = parties.view
+      .flatMap(party => stringInterning.party.tryInternalize(party).map(Int.box).toList)
+      .toArray
+    if (internedParties.length == 1)
+      cSQL"array[${internedParties.head}]::integer[]"
+    else
+      cSQL"array(select unnest(#$witnessesColumnName) intersect select unnest($internedParties::integer[]))"
+  }
 
   override def submittersArePartiesClause(
       submittersColumnName: String,
       parties: Set[Party],
+      stringInterning: StringInterning,
   ): CompositeSql = {
-    val partiesArray = parties.view.map(_.toString).toArray
-    cSQL"(#$submittersColumnName::text[] && $partiesArray::text[])"
+    val partiesArray: Array[java.lang.Integer] = parties.view
+      .flatMap(party => stringInterning.party.tryInternalize(party).map(Int.box).toList)
+      .toArray
+    cSQL"(#$submittersColumnName::integer[] && $partiesArray::integer[])"
   }
 
   override def witnessesWhereClause(
       witnessesColumnName: String,
       filterParams: FilterParams,
+      stringInterning: StringInterning,
   ): CompositeSql = {
     val wildCardClause = filterParams.wildCardParties match {
       case wildCardParties if wildCardParties.isEmpty =>
         Nil
 
       case wildCardParties =>
-        val partiesArray = wildCardParties.view.map(_.toString).toArray
-        cSQL"(#$witnessesColumnName::text[] && $partiesArray::text[])" :: Nil
+        val partiesArray: Array[java.lang.Integer] = wildCardParties.view
+          .flatMap(party => stringInterning.party.tryInternalize(party).map(Int.box).toList)
+          .toArray
+        if (partiesArray.isEmpty)
+          Nil
+        else
+          cSQL"(#$witnessesColumnName::integer[] && $partiesArray::integer[])" :: Nil
     }
     val partiesTemplatesClauses =
-      filterParams.partiesAndTemplates.iterator.map { case (parties, templateIds) =>
-        val partiesArray = parties.view.map(_.toString).toArray
-        val templateIdsArray = templateIds.view.map(_.toString).toArray
-        cSQL"( (#$witnessesColumnName::text[] && $partiesArray::text[]) AND (template_id = ANY($templateIdsArray::text[])) )"
-      }.toList
-    (wildCardClause ::: partiesTemplatesClauses).mkComposite("(", " OR ", ")")
+      filterParams.partiesAndTemplates.iterator
+        .map { case (parties, templateIds) =>
+          (
+            parties.flatMap(s => stringInterning.party.tryInternalize(s).toList),
+            templateIds,
+          )
+        }
+        .filterNot(_._1.isEmpty)
+        .filterNot(_._2.isEmpty)
```

Comment:
Yes: everything is possible since on the API real strings (party, template id) arrive, and during interning those we might end up with empty lists...ahh I just realized you are referring to the template_ids which are one PR away from here: then this will make sense, right now not.